docs/player/MemePlayer.js

```diff
@@ -224,27 +2035,30 @@ function createContentData(canvas, ctx, text, image, cb) {
     encoder = new GIF.Encoder();
     encoder.setRepeat(0);
     encoder.start();
-    renderNextFrame = function () {
-      frame = image.frames[frameIndex++];
-      encoder.setDelay(frame.delay * 10); // 1/100ths sec -> ms
-      ctx.fillStyle = "white";
-      ctx.fillRect(0, 0, frame.frame.width, frame.frame.height);
-      ctx.putImageData(frame.frame, 0, 0);
-      renderText(ctx, text, frame.frame);
-      encoder.addFrame(ctx);
-
-      if (frameIndex === image.frames.length) {
-        encoder.finish();
-        cb(null, {
-          data: btoa(encoder.stream().getData()),
-          contentType: 'image/gif'
-        });
-      } else {
-        // perf: don't bog down the event loop.
-        setTimeout(renderNextFrame, 0);
-      }
-    };
-    renderNextFrame();
+    return new Promise(function (resolve) {
+      var renderNextFrame = function () {
+        frame = image.frames[frameIndex++];
+        encoder.setDelay(frame.delay * 10); // 1/100ths sec -> ms
+        ctx.fillStyle = "white";
+        ctx.fillRect(0, 0, frame.frame.width, frame.frame.height);
+        ctx.putImageData(frame.frame, 0, 0);
+        renderText(ctx, text, frame.frame);
+        encoder.addFrame(ctx);
+
+        if (frameIndex === image.frames.length) {
+          encoder.finish();
+          resolve({
+            data: btoa(encoder.stream().getData()),
```

Comment:
update: `btoa` is now pulled from [isomorphic-base64](https://github.com/ksheedlo/isomorphic-base64)